src/main/kotlin/de/bund/digitalservice/useid/error/GlobalErrorAttributes.kt

```diff
@@ -23,9 +26,11 @@ class GlobalStaticError(
         val errorAttributes = super.getErrorAttributes(webRequest, options)
         val statusCode = errorAttributes["status"] as Int
 
-        errorAttributes["showReportEmail"] = true
-        errorAttributes["errorReportEmailLink"] = createEmailReportLink(statusCode)
-        errorAttributes["baseUrl"] = applicationProperties.baseUrl
+        val acceptedMediaType = webRequest?.getHeaderValues(HttpHeaders.ACCEPT)?.get(0)
+        if (acceptedMediaType != null && acceptedMediaType.contains(MediaType.TEXT_HTML_VALUE)) {
+            errorAttributes[SHOW_REPORT_EMAIL] = true
+            errorAttributes[ERROR_REPORT_EMAIL_LINK] = createEmailReportLink(statusCode)
+        }
```

Comment:
You could get rid of the != null check by using ?.let { } 

e.g.

webRequest?.getHeaderValues(HttpHeaders.ACCEPT)?.get(0)?.let { acceptedMediaType ->
if (acceptedMediaType.contains(MediaType.TEXT_HTML_VALUE)) {
errorAttributes[SHOW_REPORT_EMAIL] = true
errorAttributes[ERROR_REPORT_EMAIL_LINK] = createEmailReportLink(statusCode)
}
}

Reply:
Done